File: 2025-10-03.py
def check_strength(password):
    def check_length(p):

        if len(p) >= 8:
            return True
        else:
            return False

    def check_upper_lower(p):

        upper = False
        lower = False

        for char in p:
            if char.isupper():
                upper = True
            elif char.islower():
                lower = True

        return upper and lower

    def check_number(p):

        for char in p:
            if char.isdigit(): return True

        return False

    def check_special_character(p):

        must_contain = ["!", "@", "#", "$", "%", "^", "&", "*"]
        if any(symbol in p for symbol in must_contain): return True

        return False

    has_length = check_length(password)
    has_upper_lower = check_upper_lower(password)
    has_number = check_number(password)
    has_special_character = check_special_character(password)
    # rule instances

    r_count = 0

    if has_length: r_count += 1
    if has_upper_lower: r_count += 1
    if has_number: r_count += 1
    if has_special_character: r_count += 1

    if r_count < 2:
        return "weak"
    elif r_count < 4:
        return "medium"
    else:
        return "strong"

File: 2025-10-03.js
function checkStrength(password) {

    const hasLength = checkLength(password)
    const hasUpperLower = checkUpperLower(password)
    const hasNumber = checkNumber(password)
    const hasSpecialCharacter = checkSpecialCharacter(password)
    // rule instances

    let rCount = 0

    if (hasLength) {rCount ++}
    if (hasUpperLower) {rCount ++}
    if (hasNumber) {rCount ++}
    if (hasSpecialCharacter) {rCount ++}

    if (rCount < 2) {return "weak"}
    else if (rCount < 4) {return "medium"}
    else {return "strong"}
}

function checkLength(p) {
    if (p.length >= 8) {
        return true
    } return false
}

function checkUpperLower(p) {
    const
        upper = /[A-Z]/.test(p),
        lower = /[a-z]/.test(p);

    return upper && lower
}

function checkNumber(p) {
    return /[0-9]/.test(p)
}

function checkSpecialCharacter(p) {
    return /[!,@,#,$,%,^,&,*]/.test(p)
}